pangeo_forge_recipes/transforms.py

```diff
@@ -446,6 +447,33 @@ def expand(self, references: beam.PCollection) -> beam.PCollection:
         )
 
 
+@dataclass
+class WriteReference(beam.PTransform, ZarrWriterMixin):
+    """Store a singleton PCollection consisting of a ``kerchunk.combine.MultiZarrToZarr`` object.
+    :param store_name: Zarr store will be created with this name under ``target_root``.
+    :param concat_dims: Dimensions along which to concatenate inputs.
+    :param target_root: Root path the Zarr store will be created inside; ``store_name``
+      will be appended to this prefix to create a full path.
+    :param output_file_name: Name to give the output references file
+      (``.json`` or ``.parquet`` suffix).
+    """
+
+    store_name: str
+    concat_dims: List[str]
+    target_root: Union[str, FSSpecTarget, RequiredAtRuntimeDefault] = field(
```

Comment:
Great catch! I updated injections.py to add `WriteReference`. Any idea if any more is required for that?

Reply:
Can take a closer look tomorrow but off the cuff I wonder if the failure is because we need to bump the version of runner used in the integration tests

Reply:
I bumped runner to the two latest tags. First kerchunk integration test is passing! 
Getting a failure on the minio version though. 


```
tests/test_integration.py::test_python_json_configs_identical PASSED
tests/test_integration.py::test_integration[local_confpath-hrrr-kerchunk-concat-valid-time] XFAIL
tests/test_integration.py::test_integration[local_confpath-hrrr-kerchunk-concat-step] PASSED
tests/test_integration.py::test_integration[local_confpath-gpcp-from-gcs-dynamic-chunks] PASSED
tests/test_integration.py::test_integration[local_confpath-terraclimate] XFAIL
tests/test_integration.py::test_integration[local_confpath-narr-opendap] XFAIL
tests/test_integration.py::test_integration[local_confpath-noaa-oisst] PASSED
tests/test_integration.py::test_integration[local_confpath-gpcp-from-gcs] PASSED
tests/test_integration.py::test_integration[minio_confpath-hrrr-kerchunk-concat-valid-time] XFAIL
tests/test_integration.py::test_integration[minio_confpath-hrrr-kerchunk-concat-step] FAILED
```

```
raise NoCredentialsError()\nRuntimeError: botocore.exceptions.NoCredentialsError: Unable to locate credentials [while running \'Create|OpenWithKerchunk|WriteCombinedReference|Test dataset/WriteCombinedReference/WriteReference/Map(write_combined_reference)\']\n\n').returncode
```

Reply:
This is actually great, insofar as it proves the value of the `minio` test! So much better to catch this here, rather than merge this and only find out in prod! 😃

Reply:
100%

Reply:
I just rediscovered this bug since I didn't see the need to pass `remote_options` or `target_options` given we had dep injected `target_root`.

So the breaking change between `0.10.4` and what this PR merged into `main` is us removing [this context manager](https://github.com/pangeo-forge/pangeo-forge-recipes/blob/0.10.4/pangeo_forge_recipes/writers.py#L123)